src/erasure-code/ErasureCode.cc

```diff
@@ -71,14 +71,17 @@ int ErasureCode::create_rule(
   return ruleid;
 }
 
-int ErasureCode::sanity_check_k(int k, ostream *ss)
+int ErasureCode::sanity_check_k_m(int k, int m, ostream *ss)
 {
   if (k < 2) {
     *ss << "k=" << k << " must be >= 2" << std::endl;
     return -EINVAL;
-  } else {
-    return 0;
   }
+  if (m < 1) {
+    *ss << "m=" << m << " must be >= 1" << std::endl;
+    return -EINVAL;
+  }
```

Comment:
This is making layers_sanity_checks fail.

[ RUN      ] ErasureCodeLrc.layers_sanity_checks
load: jerasure the 'mapping' profile is missing from {crush-device-class=,crush-root=default,k=-1,l=-1,layers=[ ],m=-1}layers parameter has 0 which is less than the minimum of one. [ ]
m=0 must be >= 1
/home/jenkins-build/build/workspace/ceph-pull-requests/src/test/erasure-code/TestErasureCodeLrc.cc:395: Failure
      Expected: -(4095 + 9)
      Which is: -4104
To be equal to: lrc.init(profile, &cerr)
      Which is: -22